src/tbx/mod.rs

```diff
@@ -0,0 +1,410 @@
+// Copyright 2018 Manuel Holtgrewe, Berlin Institute of Health.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::ffi;
+use std::mem;
+use std::ptr;
+use std::path::Path;
+use url::Url;
+use libc;
+
+use htslib;
+
+/// A trait for a Tabix reader with a read method.
+pub trait Read: Sized {
+    /// Read next line the given `String`.
+    /// Use this method in combination with a single allocated record to avoid the reallocations
+    /// occurring with the iterator.
+    ///
+    /// # Arguments
+    ///
+    /// * `record` - the `String` to be filled
+    fn read(&mut self, record: &mut String) -> Result<(), ReadError>;
```

Comment:
What does this give you? A tabix record? Or a record from the associated file (e.g., bed)?

Reply:
I think this should be clarified now.

Reply:
Mhm, where? I still don't get what this is supposed to be. But this might be my lack of tabix knowledge. Can you explain or point me to the place where you explain it?

Reply:
No, I just forgot to `git push`...